File: src/tests/stuff/moretest/primes.py
print("primes")
limit = 10000000
sieve = [1] * limit
sieve[0] = 0
sieve[1] = 0
i = 2
while i * i < limit:
    if sieve[i] == 1:
        j = i * i
        while j < limit:
            sieve[j] = 0
            j = j + i
    i = i + 1
total = 0
i = 2
while i < limit:
    if sieve[i] == 1:
        total = total + i
    i = i + 1
print(total)

File: src/tests/stuff/moretest/primes.rs
fn main() {
    println!("primes");
    let limit = 10_000_000usize;
    let mut sieve = vec![true; limit];
    sieve[0] = false;
    sieve[1] = false;
    let mut i = 2;
    while i * i < limit {
        if sieve[i] {
            let mut j = i * i;
            while j < limit {
                sieve[j] = false;
                j += i;
            }
        }
        i += 1;
    }
    let mut total: i64 = 0;
    let mut k = 2;
    while k < limit {
        if sieve[k] {
            total += k as i64;
        }
        k += 1;
    }
    println!("{}", total);
}
